chapters/arrays.tex

```diff
@@ -1,7 +1,8 @@
 \chapter{Arrays}\label{arrays}
 
-A non-record array can be regarded as a collection collection of type compatible values, \cref{type-compatible-expressions}.
-An array of records may contain scalar record values those elements different in their dimension sizes, but apart from that they must be of the same type. Such heterogenous arrays may only be used completely, sliced as specified, or indexed.
+A non-record array can be regarded as a collection of type compatible values, \cref{type-compatible-expressions}.
+An array of records may contain scalar record values whose elements different in their dimension sizes, but apart from that they must be of the same type.
```

Comment:
There is still a verb missing in "whose elements different". It should be "elements differ" or "elements are different".